lms/djangoapps/learner_home/views.py

```diff
@@ -0,0 +1,183 @@
+    """Get settings used for platform level connections: emails, url routes, etc."""
+
+    return {
+        "supportEmail": settings.DEFAULT_FEEDBACK_EMAIL,
+        "billingEmail": settings.PAYMENT_SUPPORT_EMAIL,
+        "courseSearchUrl": marketing_link("COURSES"),
+    }
+
+
+def get_user_account_confirmation_info(user):
+    """Determine if a user needs to verify their account and related URL info"""
+
+    activation_email_support_link = (
+        configuration_helpers.get_value(
+            "ACTIVATION_EMAIL_SUPPORT_LINK", settings.ACTIVATION_EMAIL_SUPPORT_LINK
+        )
+        or settings.SUPPORT_SITE_LINK
+    )
+
+    email_confirmation = {
+        "isNeeded": not user.is_active,
+        "sendEmailUrl": activation_email_support_link,
+    }
+
+    return email_confirmation
+
+
+def get_enrollments(user, org_allow_list, org_block_list, course_limit=None):
+    """Get enrollments and enrollment course modes for user"""
+
+    course_enrollments = list(
+        get_course_enrollments(user, org_allow_list, org_block_list, course_limit)
+    )
+
+    # Sort the enrollment pairs by the enrollment date
+    course_enrollments.sort(key=lambda x: x.created, reverse=True)
+
+    # Record how many courses there are so that we can get a better
+    # understanding of usage patterns on prod.
+    monitoring_utils.accumulate("num_courses", len(course_enrollments))
+
+    # Retrieve the course modes for each course
+    enrolled_course_ids = [enrollment.course_id for enrollment in course_enrollments]
+    __, unexpired_course_modes = CourseMode.all_and_unexpired_modes_for_courses(
+        enrolled_course_ids
+    )
+    course_modes_by_course = {
+        course_id: {mode.slug: mode for mode in modes}
+        for course_id, modes in unexpired_course_modes.items()
+    }
+
+    # Construct a dictionary of course mode information
+    # used to render the course list.  We re-use the course modes dict
+    # we loaded earlier to avoid hitting the database.
+    course_mode_info = {
+        enrollment.course_id: complete_course_mode_info(
+            enrollment.course_id,
+            enrollment,
+            modes=course_modes_by_course[enrollment.course_id],
+        )
+        for enrollment in course_enrollments
+    }
+
+    return course_enrollments, course_mode_info
+
+
+def get_email_settings_info(user, course_enrollments):
+    """
+    Given a user and enrollments, determine which courses allow bulk email (show_email_settings_for)
+    and which the learner has opted out from (optouts)
+    """
+    course_optouts = Optout.objects.filter(user=user).values_list(
+        "course_id", flat=True
+    )
+
+    # only show email settings for Mongo course and when bulk email is turned on
+    show_email_settings_for = frozenset(
+        enrollment.course_id
+        for enrollment in course_enrollments
+        if (is_bulk_email_feature_enabled(enrollment.course_id))
+    )
+
+    return show_email_settings_for, course_optouts
```

Comment:
Small nit: why not have the return value be like, also the order these things are defined

Reply:
^ Is this a typo? Unclear what you're saying.

Reply:
For this, and most other design questions, the goal was "match existing behavior and names as closely as possible, refactor later".